optimizer/src/main/java/com/qubole/quark/planner/Parser.java

```diff
@@ -132,14 +130,13 @@ private String getParsedSql(RelNode relNode, SqlDialect dialect) throws SQLExcep
     return result.replace("\n", " ");
   }
 
-  public ParserResult parse(String sql) {
+  public ParserResult parse(String sql) throws SQLException {
     DataSourceSchema dataSource = this.context.getDefaultDataSource();
-    ParserResult result = new ParserResult(sql, dataSource, null, null, false);
     final AtomicBoolean foundNonQuarkScan = new AtomicBoolean(false);
     final ImmutableSet.Builder<DataSourceSchema> dsBuilder = new ImmutableSet.Builder<>();
     try {
       final SqlKind kind = getSqlParser(sql).parseQuery().getKind();
-      result = new ParserResult(stripNamespace(sql, dataSource),
+      ParserResult result = new ParserResult(stripNamespace(sql, dataSource),
```

Comment:
nit: Value being initialized here was again a fallback and can be removed now. This initialized value should never flow to return statement in current scheme of things.